### DISCUSSION TITLE
White square at the left upper corner of screen

### DISCUSSION THREAD

**Discussion by shayaniox** [i3/i3#5796] [Q&A]
Info of system:
+ Ubuntu 20.04
+ i3 4.22 (installed from [the link](https://i3wm.org/downloads/) by adding repository)

There is a white square at the left upper corner of my screen.
Can someone give a hint how to figure out what the problem is?
Please let me know if further information is required.

![i3](https://github.com/i3/i3/assets/39003982/af40c798-f16c-40d9-9e98-5f13fd6bdb7a)

**Comment by orestisfl** [comment]
Can you run xprop and click on the white square?

Does it also appear when a window is overlapping it?

What's your driver and compositor setup?

**Reply by flux77** [reply]
I'm not the OP, but I too have this issue on Debian 12. I see the white square in all workspaces. It appears to be part of the desktop background. I did not set any desktop background for i3.

i3 version: 4.22.

> Can you run xprop and click on the white square?

That white square gives the same xprop output as the rest of the black background. xprop output:

```
_NET_ACTIVE_WINDOW(WINDOW): window id # 0x1823ee4
InterpRegistry(STRING) = 
I3_LOG_STREAM_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/log-stream-socket.2193"
I3_CONFIG_PATH(UTF8_STRING) = "/home/user/.config/i3/config"
I3_PID(CARDINAL) = 2193
I3_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/ipc-socket.2193"
_NET_CLIENT_LIST(WINDOW): window id # 0x1800006, 0x1a0002c, 0x1e00006, 0x1823ee4
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0x1e00006, 0x1800006, 0x1a0002c, 0x1823ee4
_NET_DESKTOP_NAMES(UTF8_STRING) = "1", "2"
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0, 0, 0
_NET_NUMBER_OF_DESKTOPS(CARDINAL) = 2
_NET_CURRENT_DESKTOP(CARDINAL) = 1
_NET_SUPPORTED(ATOM) = _NET_SUPPORTED, _NET_SUPPORTING_WM_CHECK, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_MOVERESIZE, _NET_WM_STATE_STICKY, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_WM_STATE_MODAL, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FOCUSED, _NET_WM_STATE, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_SPLASH, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_DROPDOWN_MENU, _NET_WM_WINDOW_TYPE_POPUP_MENU, _NET_WM_WINDOW_TYPE_TOOLTIP, _NET_WM_WINDOW_TYPE_NOTIFICATION, _NET_WM_DESKTOP, _NET_WM_STRUT_PARTIAL, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_CURRENT_DESKTOP, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_NAMES, _NET_DESKTOP_VIEWPORT, _NET_ACTIVE_WINDOW, _NET_CLOSE_WINDOW, _NET_MOVERESIZE_WINDOW
_NET_WM_NAME(UTF8_STRING) = "i3"
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x800064
XIM_SERVERS(ATOM) = @server=ibus
AT_SPI_BUS(STRING) = "unix:path=/run/user/1000/at-spi/bus_0,guid=539f77a575e679918aad2859656aff11"
GDK_VISUALS(INTEGER) = 1444, 1946
SUPERSCRIPT_X(CARDINAL) = 
RESOURCE_MANAGER(STRING) = "*customization:\t-color\n"
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 2
```

> Does it also appear when a window is overlapping it?

No.

> What's your driver and compositor setup?

VGA compatible controller: Intel Corporation Skylake GT2 [HD Graphics 520]. Kernel driver in use: i915.

```
$ echo $XDG_SESSION_TYPE                                                      
x11
```

No compositor.

**Reply by shayaniox** [reply]
The `xprop` output:
```
_NET_ACTIVE_WINDOW(WINDOW): window id # 0xe00007
PULSE_COOKIE(STRING) = "7324af6cc569866f68255e9ed8e6f3c323aed376b7fbfa8893cec654bffed98e73bdfa91dfbda8d650a0233ab62190f581381e5810200530d57e13eec816558a85c54cc68c430baaa2925ad7ffcda211d16cd7c4cb233f7ba6ba3c9557fcec3a3a6eb21071e08963ed4d2dd4beba1d03ecf59245bf38c5b54d3581f883de97b07295573dacfd9dc73a80e9a987ba875b98c270b3333769daebd9729f0584ca27de54d5c65c9815fe9a4add03d66087d17f4ac913441af761da21d13f829067c2abac20a71de7b02d8761705333198956ebf41379c048654a1c4abcba11808b21c228bf7902788395b03c2a67052f22ca77f5c2f72cfe211d8b9c3e68bf7e02fd"
PULSE_SERVER(STRING) = "{5f1e7d727a45444aac74d6fe417bf284}unix:/run/user/1000/pulse/native"
PULSE_ID(STRING) = "1000@5f1e7d727a45444aac74d6fe417bf284/47593"
GDK_VISUALS(INTEGER) = 43, 130
AT_SPI_BUS(STRING) = "unix:abstract=/tmp/dbus-0zWAFwK4As,guid=fd97225296a36c79c387f068656cd5b3"
_MIT_SCREEN_SAVER_ID(PIXMAP): pixmap id # 0x600000
I3_LOG_STREAM_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/log-stream-socket.47475"
I3_CONFIG_PATH(UTF8_STRING) = "/home/shayan/.config/i3/config"
I3_PID(CARDINAL) = 47475
I3_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/ipc-socket.47475"
_NET_CLIENT_LIST(WINDOW): window id # 0xe00007
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0xe00007
_NET_DESKTOP_NAMES(UTF8_STRING) = "1"
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0
_NET_NUMBER_OF_DESKTOPS(CARDINAL) = 1
_NET_CURRENT_DESKTOP(CARDINAL) = 0
_NET_SUPPORTED(ATOM) = _NET_SUPPORTED, _NET_SUPPORTING_WM_CHECK, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_MOVERESIZE, _NET_WM_STATE_STICKY, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_WM_STATE_MODAL, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FOCUSED, _NET_WM_STATE, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_SPLASH, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_DROPDOWN_MENU, _NET_WM_WINDOW_TYPE_POPUP_MENU, _NET_WM_WINDOW_TYPE_TOOLTIP, _NET_WM_WINDOW_TYPE_NOTIFICATION, _NET_WM_DESKTOP, _NET_WM_STRUT_PARTIAL, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_CURRENT_DESKTOP, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_NAMES, _NET_DESKTOP_VIEWPORT, _NET_ACTIVE_WINDOW, _NET_CLOSE_WINDOW, _NET_MOVERESIZE_WINDOW
_NET_WM_NAME(UTF8_STRING) = "i3"
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x20003f
SUPERSCRIPT_X(CARDINAL) = 
RESOURCE_MANAGER(STRING) = "*.background:\t#282A36\n*.color0:\t#000000\n*.color1:\t#FF5555\n*.color10:\t#5AF78E\n*.color11:\t#F4F99D\n*.color12:\t#CAA9FA\n*.color13:\t#FF92D0\n*.color14:\t#9AEDFE\n*.color15:\t#E6E6E6\n*.color2:\t#50FA7B\n*.color3:\t#F1FA8C\n*.color4:\t#BD93F9\n*.color5:\t#FF79C6\n*.color6:\t#8BE9FD\n*.color7:\t#BFBFBF\n*.color8:\t#4D4D4D\n*.color9:\t#FF6E67\n*.foreground:\t#F8F8F2\n*customization:\t-color\n"
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us,ir", ",pes_keypad", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 2
```

> Does it also appear when a window is overlapping it?

No, actually any window overlaps the square.

> What's your driver and compositor setup?

I'm using pure `i3` with no compositor or modified config.
If it helps, I use `nvidia-525` Xorg driver.

@orestisfl

**Reply by orestisfl** [reply]
Thank you, can either (or both) of you try to reproduce with an older (4.21, 4.20) and newer (4.23) version?

**Reply by shayaniox** [reply]
Currently, there is only the version `4.22` available on `i3` repository for Ubuntu 20.04 and I didn't find any official document on installing the wm from source. Could you provide a link to installation from source to avoid errors on installation?
It's good to mention that I have installed `i3` on `debian 12` and `ubuntu 20.04` virtual machines and everything is fine in the machines.

**Reply by orestisfl** [reply]
we don't really have documentation on how to build i3 from source, here's some rough instructions for ubuntu:

```shell
sudo apt-get update
sudo apt-get install -y git build-essential meson libx11-dev libxcb-util0-dev libxcb-keysyms1-dev libxcb-xinerama0-dev libxcb-randr0-dev libxcb-icccm4-dev libxcb-cursor-dev libxcb-xrm-dev libxcb-xkb-dev libxcb-shape0-dev libxkbcommon-dev libxkbcommon-x11-dev pkg-config libev-dev libyajl-dev libpcre2-dev libstartup-notification0-dev libcairo2-dev libpango1.0-dev libpod-simple-perl
git clone https://github.com/i3/i3
cd i3
meson build
ninja -C build
```

verify that the built version is correct:
```shell
$ build/i3 --version
i3 version 4.23-7-g69f68dcd © 2009 Michael Stapelberg and contributors
```

Then `export PATH="$HOME/i3/build:$PATH"` in your bashrc/zshrc/xinitrc, log-out and log-in. Run `i3 --version` again to verify.

Then, to check with an older version:
```shell
cd i3
git checkout 4.21
ninja -C build
build/i3 --version
```
log-out and log-in

**Reply by shayaniox** [reply]
I followed your instructions and built versions: `4.21`, `4.21.1`, `4.23` and for all of them the issue persists.
Is there any ways to check if there is a conflicting package with `i3`?

**Reply by orestisfl** [reply]
To potentially cross-out i3 from the reason this is happening can you check if you see the same square in the awesomewm and openbox window managers?

**Reply by shayaniox** [reply]
I tested the `awesomewm` with solid color background and the white square didn't exist. then uninstalled `awesomewm` and tested the `i3` again and nothing was changed.

**Reply by shayaniox** [reply]
I don't have `.xinitrc` and have `.Xresources`. does it help to share the `.Xresources` content?

**Reply by orestisfl** [reply]
can you try openbox as well?

> I don't have .xinitrc and have .Xresources. does it help to share the .Xresources content?

Does xresources affect your issue? Does temporarily moving the file and restarting your session change anything?

**Reply by shayaniox** [reply]
The openbox doesn't have the issue.
Also removed the `.Xresources` and it changed nothing.

**Reply by shayaniox** [reply]
@orestisfl I think the output of the logs may help
https://file.io/Q57q1nfWCgwi

**Reply by orestisfl** [reply]
The log file doesn't exist anymore, see https://i3wm.org/docs/debugging.html#_obtaining_the_debug_logfile for instructions.

**Reply by frbl** [reply]
I have the exact same. Issue, and I think it has to do with some sort of task icon it wants to show there. If you open op chrome and run it in the background, the white square changes to the chrome logo

**Reply by yannxaver** [reply]
I wanted to ask if anyone has been able to remove the white square in the meantime?

**Reply by Petar0000** [reply]
> I wanted to ask if anyone has been able to remove the white square in the meantime?

I managed to remove it by setting a background image. If you want to keep it black, just set the background to a solid black image
